CONTRIBUTING.md

```diff
@@ -32,41 +59,64 @@ source venv/bin/activate
 
 ### Testing
 
-To run tests, run the following
+To run tests, first build the charm as described above, then run the following
 
 ```shell
 tox -e format       # update your code according to linting rules
 tox -e lint         # code style
 tox -e unit         # unit tests
-tox -m integration  # integration tests, running on juju 2.
+tox -m integration  # integration tests, running on juju 3.
 tox                 # runs 'format', 'lint', and 'unit' environments
 ```
 
-Integration tests can be run for separate areas of functionality:
+Integration tests can be run for separate files:
 
 ```shell
-tox -e charm-integration      # basic charm integration tests
-tox -e tls-integration        # TLS-specific integration tests
-tox -e client-integration     # Validates the `opensearch-client` integration
-tox -e ha-integration         # HA tests
-tox -e h-scaling-integration  # HA tests specific to horizontal scaling
+tox -e integration -- tests/integration/ha/test_storage.py
 ```
 
-If you're running tests on juju 3, run the following command to change libjuju to the correct version:
+#### Running different major versions of Juju
+
+For integration tests, libjuju must be in-sync with the target juju version.
+Make sure that the version of libjuju installed is compatible with the bootstrapped
+controller version. If not, update it with:
 
 ```shell
-export LIBJUJU_VERSION_SPECIFIER="==3.3.0.0"
+poetry add --lock --group integration juju@<YOUR CHOSEN VERSION>
 ```
 
-## Build charm
+#### FOR DEVELOPMENT ONLY: Testing Backups In Your Local Machine
+
+Backup testing installs microceph and, optionally, can run on AWS and GCP object stores.
+For that, pass the access / secret / service account information as env. variables.
+
+To run the test only against microceph:
+
+```shell
+tox -e integration -- tests/integration/ha/test_backups.py --group='microceph' # test backup service for microceph
+```
 
-Build the charm in this git repository using:
+And against public clouds + microceph:
 
 ```shell
-charmcraft pack
+SECRETS_FROM_GITHUB=$(cat <path-to>/credentials.json) tox -e integration -- tests/integration/ha/test_backups.py
 ```
 
-### Deploy
+Where, for AWS only, `credentials.json` should look like (the `IGNORE` is necessary to remove GCP):
+```python
+{ "AWS_ACCESS_KEY": ..., "AWS_SECRET_KEY": ..., "GCP_SECRET_KEY": "IGNORE" }
+```
+
+For GCP only, some additional information is needed. The pytest script runs with `boto3` and needs `GCP_{ACCESS,SECRET}_KEY` to clean up the object store; whereas OpenSearch uses the `service account` json file.
+
+```python
+{ "AWS_SECRET_KEY": "IGNORE", "GCP_ACCESS_KEY": ..., "GCP_SECRET_KEY": ..., "GCP_SERVICE_ACCOUNT": ... }
```

Comment:
I'm not a fan of the `"IGNORE"`, we shouldn't have to include other GCP related entries for AWS and vice-versa